packages/server/lib/modes/run.js

```diff
@@ -1009,10 +1010,56 @@ module.exports = {
     return openProject.launch(browser, spec, browserOpts)
   },
 
-  navigateToNextSpec (spec) {
+  async navigateToNextSpec (options) {
     debug('navigating to next spec')
+    const { browser, spec, writeVideoFrame, setScreenshotMetadata, project, screenshots, projectRoot, onError } = options
+
+    const browserOpts = getDefaultBrowserOptsByFamily(browser, project, writeVideoFrame, onError)
+
+    browserOpts.automationMiddleware = {
+      onBeforeRequest (message, data) {
+        if (message === 'take:screenshot') {
+          return setScreenshotMetadata(data)
+        }
+      },
+      onAfterResponse: (message, data, resp) => {
```

Comment:
Is this new logic to take additional screenshots or are we replacing existing logic?

Reply:
Good catch. This is actually a hold over of something I tried early on. It can all be removed.